config/hotreloadable.go

```diff
@@ -1,14 +1,74 @@
 package config
 
-import "time"
+import (
+	"strings"
+	"sync"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
-func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
-	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
+// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
+// e.g. asking for the same keys but in a different order can result in a different value to be returned
+//
```

Comment:
**Minor:** please use a separate paragraph for the warning so that it is easier to notice
```suggestion
//
// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
// e.g. asking for the same keys but in a different order can result in a different value to be returned
//
```

Reply:
God bless the regex replace feature :sunglasses: